pyfr/plugins/solutionwriter.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolutionWriterPlugin(BasePlugin, H5Writer):
+    name = 'solutionwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, *args, **kwargs):
+        BasePlugin.__init__(self, intg, cfgsect)
```

Comment:
Problem of `EulerElements` is not the exact same situation, i think, because `__init__` is implemented only by the `BaseElements` class. 

I believe you are referring to the `EulerElements::set_backend` . 
Here the issue does not arise because 
1. The `BaseFluidElements` call does not implement `set_backend`, so is passed up via `BaseAdvectionElements` and finally `BaseElements` so the order is perfectly clear,
2. `object` does not implement set_backend, so no problem differing function signatures.

Composition, however,  will work without any problem. Will switch to that.